IFTPipeline.jl/src/feature-extraction.jl

```diff
@@ -107,3 +112,75 @@ function extractfeatures_single(;
     FileIO.save(output, props)
     return nothing
 end
+
+"""
+    load_labeled_img(path)
+
+Load an unsigned integer image from a file.
+
+See also: save_labeled_img
+"""
+function load_labeled_img(path::AbstractString)
+    image = FileIO.load(path)
+    image_reinterpreted = convert_uint_from_gray(image)
+    return image_reinterpreted
+end
+
+"""
+    save_labeled_img(image, path)
+
+Save an unsigned integer image to an image file.
+
+See also: load_labeled_img
+"""
+function save_labeled_img(image::AbstractArray{T} where {T <: Union{UInt8, UInt16, UInt32, UInt64}}, path::AbstractString)
+    image_reinterpreted = convert_gray_from_uint(image)
+    FileIO.save(path, image_reinterpreted)
+    return path
+end
+
+"""
+    convert_gray_from_uint(image)
+
+Convert an image from an unsigned integer format into a fixed-point Gray format.
+
+See also: convert_uint_from_gray
+"""
+
+function convert_gray_from_uint(image::AbstractArray{T} where {T <: Union{UInt8, UInt16, UInt32, UInt64}})
+    if eltype(image) === UInt8
+        target_type = N0f8
+    elseif eltype(image) === UInt16
+        target_type = N0f16
+    elseif eltype(image) === UInt32
+        target_type = N0f32
+    elseif eltype(image) === UInt64
+        target_type = N0f64
+    end
+    image_reinterpreted  = Gray.(reinterpret.(target_type, image))
+    return image_reinterpreted
+end
```

Comment:
You can use the reference to the eltype of `image`, `T`, within the function. And because there is a counterpart to `convert_gray_from_uint`, it might be better to use the mapping pattern here and reuse the inverted mapping there.

Reply:
This is updated in a following PR.